From ff978df959937bef6c03e55accfcc1667e53e231 Mon Sep 17 00:00:00 2001
From: Daniel Caires <DanielMarques.Caires@windriver.com>
Date: Wed, 31 Jul 2024 07:55:37 -0300
Subject: [PATCH] Expose configurable Job TTL on values

Change the ttlSecondsAfterFinished for the Jobs in helm-toolkit
and expose its configuration to the values file of all
Helm-charts used by STX-Openstack.

The default value for the TTL was set at 5400 (1h30m).

Signed-off-by: Daniel Caires <DanielMarques.Caires@windriver.com>
---
 aodh/values.yaml       | 1 +
 barbican/values.yaml   | 3 +++
 ceilometer/values.yaml | 3 +++
 cinder/values.yaml     | 1 +
 glance/values.yaml     | 3 +++
 heat/values.yaml       | 1 +
 horizon/values.yaml    | 3 +++
 ironic/values.yaml     | 3 +++
 keystone/values.yaml   | 1 +
 neutron/values.yaml    | 3 +++
 nova/values.yaml       | 1 +
 placement/values.yaml  | 3 +++
 12 files changed, 26 insertions(+)

diff --git a/aodh/values.yaml b/aodh/values.yaml
index 5da6eb61..7aeb99f3 100644
--- a/aodh/values.yaml
+++ b/aodh/values.yaml
@@ -67,6 +67,7 @@ images:
       - image_repo_sync
 
 jobs:
+  ttlSecondsAfterFinished: 5400
   alarms_cleaner:
     # daily
     cron: "0 */24 * * *"
diff --git a/barbican/values.yaml b/barbican/values.yaml
index 9bea5035..720a615f 100644
--- a/barbican/values.yaml
+++ b/barbican/values.yaml
@@ -55,6 +55,9 @@ images:
       - dep_check
       - image_repo_sync
 
+jobs:
+  ttlSecondsAfterFinished: 5400
+
 pod:
   security_context:
     barbican:
diff --git a/ceilometer/values.yaml b/ceilometer/values.yaml
index 3e8d477b..7e01bcc1 100644
--- a/ceilometer/values.yaml
+++ b/ceilometer/values.yaml
@@ -1918,6 +1918,9 @@ endpoints:
       http:
         default: 15672
 
+jobs:
+  ttlSecondsAfterFinished: 5400
+
 pod:
   affinity:
     anti:
diff --git a/cinder/values.yaml b/cinder/values.yaml
index 0496c2cc..9c40eef7 100644
--- a/cinder/values.yaml
+++ b/cinder/values.yaml
@@ -69,6 +69,7 @@ images:
       - image_repo_sync
 
 jobs:
+  ttlSecondsAfterFinished: 5400
   volume_usage_audit:
     cron: "5 * * * *"
     starting_deadline: 600
diff --git a/glance/values.yaml b/glance/values.yaml
index ed8fd754..9c9dd283 100644
--- a/glance/values.yaml
+++ b/glance/values.yaml
@@ -777,6 +777,9 @@ endpoints:
       ingress:
         default: 80
 
+jobs:
+  ttlSecondsAfterFinished: 5400
+
 pod:
   security_context:
     glance:
diff --git a/heat/values.yaml b/heat/values.yaml
index 5a0e4cba..77397b9e 100644
--- a/heat/values.yaml
+++ b/heat/values.yaml
@@ -66,6 +66,7 @@ images:
       - image_repo_sync
 
 jobs:
+  ttlSecondsAfterFinished: 5400
   engine_cleaner:
     cron: "*/5 * * * *"
     starting_deadline: 600
diff --git a/horizon/values.yaml b/horizon/values.yaml
index 7aa8519d..fd68fc86 100644
--- a/horizon/values.yaml
+++ b/horizon/values.yaml
@@ -1090,6 +1090,9 @@ dependencies:
         - endpoint: internal
           service: dashboard
 
+jobs:
+  ttlSecondsAfterFinished: 5400
+
 pod:
   security_context:
     horizon:
diff --git a/ironic/values.yaml b/ironic/values.yaml
index 39643ae8..a1faf966 100644
--- a/ironic/values.yaml
+++ b/ironic/values.yaml
@@ -561,6 +561,9 @@ endpoints:
       metrics:
         default: 24220
 
+jobs:
+  ttlSecondsAfterFinished: 5400
+
 pod:
   affinity:
     anti:
diff --git a/keystone/values.yaml b/keystone/values.yaml
index 2eea531d..486f38f1 100644
--- a/keystone/values.yaml
+++ b/keystone/values.yaml
@@ -420,6 +420,7 @@ pod:
             periodSeconds: 60
             timeoutSeconds: 15
 jobs:
+  ttlSecondsAfterFinished: 5400
   fernet_setup:
     user: keystone
     group: keystone
diff --git a/neutron/values.yaml b/neutron/values.yaml
index 9f8666fc..d6962d6e 100644
--- a/neutron/values.yaml
+++ b/neutron/values.yaml
@@ -352,6 +352,9 @@ dependencies:
         - endpoint: internal
           service: local_image_registry
 
+jobs:
+  ttlSecondsAfterFinished: 5400
+
 pod:
   use_fqdn:
     neutron_agent: true
diff --git a/nova/values.yaml b/nova/values.yaml
index 0d964c53..d8194eea 100644
--- a/nova/values.yaml
+++ b/nova/values.yaml
@@ -91,6 +91,7 @@ images:
       - image_repo_sync
 
 jobs:
+  ttlSecondsAfterFinished: 5400
   # NOTE(portdirect): When using cells new nodes will be added to the cell on the hour by default.
   # TODO(portdirect): Add a post-start action to nova compute pods that registers themselves.
   cell_setup:
diff --git a/placement/values.yaml b/placement/values.yaml
index 35f4849c..2df89191 100644
--- a/placement/values.yaml
+++ b/placement/values.yaml
@@ -298,6 +298,9 @@ endpoints:
         public: 80
         service: 8778
 
+jobs:
+  ttlSecondsAfterFinished: 5400
+
 pod:
   security_context:
     placement:
-- 
2.45.2

